lib/fastlane/plugin/wpmreleasetoolkit/helper/git_helper.rb

```diff
@@ -124,7 +124,18 @@ def self.list_local_tags(matching: '*')
       # @param [Bool] delete_on_remote If true, will also delete the tag from the remote. Otherwise, it will only be deleted locally.
       #
       def self.delete_tags(tag_names, delete_on_remote: false)
-        Action.sh('git', 'tag', '-d', *tag_names)
+
+        tag_names = [tag_names] unless tag_names.is_a? Array
+
+        g = Git.open(Dir.pwd)
+        local_tag_names = g.tags.map do |tag|
+          tag.name
+        end
+
+        tag_names.each do |tag|
+          Action.sh('git', 'tag', '-d', *tag) if local_tag_names.include? tag
```

Comment:
(1) Since you already started to use the `Git` gem above (👍 ), can't we use `g` to delete the tags here, instead of `Action.sh`?

(2) Any benefit of iterating over the array only to pass one tag on each iteration, instead of passing all the tags directly in one call?
```ruby
local_tags_to_delete = tag_names & local_tag_names # '&' is the 'set/array intersection' operator here
Action.sh('git', 'tag', '-d', *local_tags_to_delete) # Or even better if you find in the gem doc a way to do that using the gem, didn't look it up
```

Reply:
1. Good call – done in d3d0d673deebb9a288050efa0da9e92bdb279403.
2. Unfortunately the `git` gem only supports one-at-a-time, but it shouldn't be a huge deal as there are typically few tags to delete.

Also agree with the variable naming – it's simplified quite a bit now.